doi-transformer/src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java

```diff
@@ -472,62 +529,191 @@ public void toPublicationPreservesOrcidWhenAuthorHasOrcid() throws InvalidIssnEx
     }
 
     @Test
-    @DisplayName("toPublication sets multiple affiliation labels when author has more affiliations name")
-    public void toPublicationSetsMultipleAffiliationLabelWhenAuthorHasMultipleAffiliationNameDuplicate()
-            throws InvalidIssnException {
-        setAuthorWithMultipleAffiliations(sampleInputDocument);
-        Publication actualDocument = toPublication(sampleInputDocument);
+    @DisplayName("toPublication sets PublicationContext to Book when crossref-type is book")
+    public void toPublicationSetsPublicationContextToBookWhjenCrossrefDocumentHasTypeBook()
+            throws InvalidIssnException, InvalidIsbnException {
+        CrossRefDocument sampleInputDocumentBook = createSampleDocumentBook();
+        Publication actualDocument = toPublication(sampleInputDocumentBook);
+        var publicationContext =  actualDocument.getEntityDescription().getReference().getPublicationContext();
 
-        List<Contributor> contributors = actualDocument.getEntityDescription().getContributors();
-        List<Organization> organisations =
-                contributors.stream()
-                .map(Contributor::getAffiliations)
-                .filter(Objects::nonNull)
-                .flatMap(Collection::stream)
+        assertTrue(publicationContext instanceof Book);
+    }
+
+    @Test
+    @DisplayName("toPublication sets ISBN when crossref-type is book")
+    public void toPublicationSetsIsbnWhenCrossrefDocumentIsTypeBook()
+            throws InvalidIssnException, InvalidIsbnException {
+        CrossRefDocument sampleInputDocumentBook = createSampleDocumentBook();
+
+        IsxnType type = IsxnType.PRINT;
+        Isxn printIsbnA = sampleIsxn(type, VALID_ISBN_A);
+        Isxn printIsbnB = sampleIsxn(type, VALID_ISBN_B);
+        List<Isxn> isbns = Arrays.asList(printIsbnA, printIsbnB);
+        sampleInputDocumentBook.setIsbnType(isbns);
+
+        Publication actualDocument = toPublication(sampleInputDocumentBook);
+        Book actualPublicationContext = (Book) actualDocument.getEntityDescription()
+                .getReference()
+                .getPublicationContext();
+        var actualIsbnList = actualPublicationContext.getIsbnList();
+
+        List<String> poolOfExpectedValues = isbns.stream()
+                .map(Isxn::getValue)
+                .map(ISBN_VALIDATOR::validate)
                 .collect(Collectors.toList());
+        assertTrue(actualIsbnList.containsAll(poolOfExpectedValues));
```

Comment:
should compare if the two collections are equal probably. Make them into sets and compare for equality?

Reply:
Comparing sets..